nautobot/core/views/generic.py

```diff
@@ -1278,18 +1278,53 @@ def get_required_permission(self):
     def get(self, request):
         return redirect(self.get_return_url(request))
 
+    def handle_bulk_delete_all(self, request, **kwargs):
+        model = self.queryset.model
+        self.template_name = "generic/object_bulk_destroy_all.html"
+
+        if self.filterset is not None:
+            # We take this approach because filterset.qs has already applied .distinct(),
+            # and performing a .delete directly on a queryset with .distinct applied is not allowed.
+            pk_list = self.filterset(request.GET, model.objects.only("pk")).qs.values_list("pk")
+            queryset = self.queryset.filter(pk__in=pk_list)
```

Comment:
This approach is likely to still have issues at scale (e.g. if `pk_list` evaluates to 100000 records). We should be able to pass around the filter params (`request.GET` in the base view code) and pass those to the filterset/queryset instead.